v2/spanner-change-streams-to-sharded-file-sink/src/test/java/com/google/cloud/teleport/v2/templates/SpannerChangeStreamToGcsMultiShardIT.java

```diff
@@ -0,0 +1,224 @@
+            .set("SingerId")
+            .to(singerId)
+            .set("FirstName")
+            .to(firstName)
+            .set("migration_shard_id")
+            .to(shardId)
+            .build();
+    spannerResourceManager.write(m);
+  }
+
+  @Test
+  public void testMultiShardsRecordWrittenToGcs()
+      throws IOException, java.lang.InterruptedException {
+    // Construct a ChainedConditionCheck with below stages.
+    // 1. Wait for the metadata table to have the start time of reader job
+    // 2. Write 2 records per shard to Spanner
+    // 3. Wait on GCS to have the files
+    // 4. Match the PK in GCS with the PK written to Spanner
+    ChainedConditionCheck conditionCheck =
+        ChainedConditionCheck.builder(
+                List.of(
+                    SpannerRowsCheck.builder(
+                            spannerMetadataResourceManager, "spanner_to_gcs_metadata")
+                        .setMinRows(1)
+                        .setMaxRows(1)
+                        .build()))
+            .build();
+    // Wait for conditions
+    PipelineOperator.Result result =
+        pipelineOperator()
+            .waitForCondition(createConfig(jobInfo, Duration.ofMinutes(10)), conditionCheck);
+    // Assert Conditions
+    assertThatResult(result).meetsConditions();
+    // Perform writes to Spanner
+    writeSpannerDataForSingers(2, "two", "testShardB");
+    writeSpannerDataForSingers(3, "three", "testShardB");
+    writeSpannerDataForSingers(4, "four", "testShardC");
+    writeSpannerDataForSingers(5, "five", "testShardC");
+
+    // Assert file present in GCS with the needed data
+    assertFileContentsInGCSForMultipleShards();
+  }
+
+  private void assertFileContentsInGCSForMultipleShards() {
+    ChainedConditionCheck conditionCheck =
+        ChainedConditionCheck.builder(
+                List.of(
+                    GCSArtifactsCheck.builder(
+                            gcsResourceManager, "output/testShardB/", Pattern.compile(".*\\.txt$"))
+                        .setMinSize(1)
+                        .setMaxSize(2)
+                        .build(),
+                    GCSArtifactsCheck.builder(
+                            gcsResourceManager, "output/testShardC/", Pattern.compile(".*\\.txt$"))
+                        .setMinSize(1)
+                        .setMaxSize(2)
+                        .build()))
+            .build();
+
+    PipelineOperator.Result result =
+        pipelineOperator()
+            .waitForCondition(createConfig(jobInfo, Duration.ofMinutes(6)), conditionCheck);
+
+    // Assert Conditions
+    assertThatResult(result).meetsConditions();
+
+    List<Artifact> artifactsShardB =
+        gcsResourceManager.listArtifacts("output/testShardB/", Pattern.compile(".*\\.txt$"));
+    List<Artifact> artifactsShardC =
+        gcsResourceManager.listArtifacts("output/testShardC/", Pattern.compile(".*\\.txt$"));
+    assertThatArtifacts(artifactsShardB).hasFiles();
```

Comment:
Is this check needed after the GCSArtifacts check - which itself checks the count of files?